presto-docs/src/main/sphinx/connector/kudu.rst

```diff
@@ -0,0 +1,494 @@
+==============
+Kudu Connector
+==============
+
+The kudu connector allows querying, inserting and deleting data in `Apache Kudu`_
+
+.. _Apache Kudu: https://kudu.apache.org/
+
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+
+Compatibility
+-------------
+
+Connector is compatible with all Apache Kudu versions starting from 1.0.
+
+If the connector uses features that are not available on the target server, an error will be returned.
+Apache Kudu 1.7.0 is currently used for testing.
+
+
+Configuration
+-------------
+
+To configure the Kudu connector, create a catalog properties file
+``etc/catalog/kudu.properties`` with the following contents,
+replacing the properties as appropriate:
+
+  .. code-block:: none
+
+       connector.name=kudu
+
+       ## List of Kudu master addresses, at least one is needed (comma separated)
+       ## Supported formats: example.com, example.com:7051, 192.0.2.1, 192.0.2.1:7051,
+       ##                    [2001:db8::1], [2001:db8::1]:7051, 2001:db8::1
+       kudu.client.master-addresses=localhost
+
+       ## Optional restriction of tablets for specific tenant.
+       ## If a tenant is set, only Kudu tablets starting with `<tenant>.` will
+       ## be visible in Presto
+       #kudu.session.tenant=mytenant
+
+       #######################
+       ### Advanced Kudu Java client configuration
+       #######################
+
+       ## Default timeout used for administrative operations (e.g. createTable, deleteTable, etc.)
+       #kudu.client.defaultAdminOperationTimeout = 30s
+
+       ## Default timeout used for user operations
+       #kudu.client.defaultOperationTimeout = 30s
+
+       ## Default timeout to use when waiting on data from a socket
+       #kudu.client.defaultSocketReadTimeout = 10s
+
+       ## Disable Kudu client's collection of statistics.
+       #kudu.client.disableStatistics = false
+
+
+Querying Data
+-------------
+
+A Kudu table named ``mytable`` is available in Presto as table
+``kudu.default.mytable``. A Kudu table containing a dot is considered as
+a schema/table combination, e.g. ``dev.mytable`` is mapped to the Presto
+table \`kudu.dev.mytable. Only Kudu table names in lower case are
+currently supported.
+
+Before using any tablets, it is needed to create the default schema,
+e.g.
+
+  .. code:: sql
+
+    CREATE SCHEMA default;
+
+Example
+~~~~~~~
+
+-  Create default schema if needed:
+
+  .. code:: sql
+
+    CREATE SCHEMA IF NOT EXISTS default;
+
+-  Now you can use any Kudu table, if it is lower case and contains no
+   dots.
```

Comment:
why not to use `String::toLowerCase` for names so even tables with upper case letters will be accessible from Presto.